blockchain/chain_sync/src/sync.rs

```diff
@@ -78,7 +78,7 @@ impl Default for SyncConfig {
 /// messages to be able to do the initial sync.
 pub struct ChainSyncer<DB, TBeacon, V, M> {
     /// State of general `ChainSync` protocol.
-    state: ChainSyncState,
+    state: Arc<Mutex<ChainSyncState>>,
```

Comment:
is mutex over rwlock required?

Reply:
It is not, but does `RwLock` have any benefits if we're not expecting concurrent reads?

Reply:
Yeah I just was asking because there doesn't seem to be a reason to have the additional constraint (just thinking for future proofing)

It's definitely fine if this stays as is, this will probably be replaced in the future anyway